universum/configuration_support.py

```diff
@@ -12,6 +15,203 @@
 
 skip_attributes = {"children", "critical", "skip_numbering_level"}
 
+ProjectConfigKeyType = Literal["name", "command", "environment", "artifacts", "report_artifacts",
+                               "artifact_prebuild_clean", "directory", "critical", "background", "finish_background",
+                               "code_report", "pass_tag", "fail_tag", "if_env_set"]
+
+
+class ProjectConfig(TypedDict, total=False):
```

Comment:
1. Since users are not going to use this class, it has to be removed from the documentation.
Alternatively, we can change the approach to creating configs and let users to use this class.
However, this would be breaking change.
2. If we are discussing breaking changes, maybe it would be better to move away from a dictionary.
As an option, Variations can convert dictionary to the class automatically, but accept both variants.

Reply:
as temporary solution, I kept inheritance from dict, but I intend to remove that dependency in this PR, after I rewrite all other modules to use properties instead of accessing dict keys. I already updated Variations to accept either dict or ProjectConfiguration - which is the only change that would require new test case I think.